[ML] Update the voyageai integration

fzowl opened the pull request:
- correcting the multimodal integration
  - adding the contextual model support
  - refactoring

fzowl committed:
VoyageAI embeddings and rerank:
 - embeddings works, tested
 - initial rerank code

What's missing:
 - unit and integration tests
 - rerank request/response mapping and verification

fzowl committed:
VoyageAI embeddings and rerank:
 - embeddings works, tested
 - rerank works, tested (https://www.elastic.co/search-labs/blog/elasticsearch-cohere-rerank)

What's missing:
 - unit and integration tests

fzowl committed:
VoyageAI embeddings and rerank:
 - embeddings works, tested
 - rerank works, tested (https://www.elastic.co/search-labs/blog/elasticsearch-cohere-rerank)

What's missing:
 - unit and integration tests

fzowl committed:
Adding initial tests
Moving dimensions to ServiceSettings

fzowl committed:
Correcting due to comments

fzowl committed:
Adding BIT support

fzowl committed:
Initial tests

fzowl committed:
More tests

fzowl committed:
More tests/corrections

fzowl committed:
Removing warnings

fzowl committed:
Further tests

fzowl committed:
Adding changelog and correcting TransportVersions

fzowl committed:
Spotless tests

fzowl committed:
Changes due to the comments

fzowl committed:
Changes due to the comments

fzowl committed:
Adding VoyageAI's v3.5 models

fzowl committed:
Adding VoyageAI's v3.5 models

fzowl commented:
@davidkyle Can you please take a look?

davidkyle reviewed (commented):
@fzowl there has been a refactoring since the original PR #122134. All the VoyageAI related request and response classes have been moved from the `org.elasticsearch.xpack.inference.external` package to `org.elasticsearch.xpack.inference.services.voyageai`.  It looks like git has recreated all the old classes in `org.elasticsearch.xpack.inference.external` 

Please can you try to apply the changes to the classes in `org.elasticsearch.xpack.inference.services.voyageai`. It may be easier to close this PR and open another with the last few commits from this PR applied - e778e3619a071fe294e0397ecd5338ead59195fc and b7351e791161744134c364ee62efa7de3a8a517f?

Thanks

theletterf committed:
Fix irregular spaces (#137014)

* Fix irregular spaces

* Update analysis-keyword-repeat-tokenfilter.md

* Update search-suggesters.md

* Update search-profile.md

fzowl committed:
Update VoyageAI integration

- Add multimodal embeddings support with VoyageAIMultimodalEmbeddings* classes
- Add contextual embeddings support with VoyageAIContextualEmbeddings* classes
- Refactor text embeddings: move to embeddings/text subdirectory for better organization
- Remove old embedding classes from embeddings root directory
- Update action creators and visitors to handle new embedding types
- Update imports in InferenceNamedWriteablesProvider
- Add corresponding request/response entities for multimodal and contextual embeddings
- Update unit tests for all new and modified components

fzowl committed:
Add text embeddings test files and fix model API calls

- Move text embeddings test files to text/ subdirectory
- Fix package declarations in test files
- Update VoyageAIEmbeddingsResponseEntityTests to use correct API methods
- Change .service() to .getService() and add .getConfigurations()
- NOTE: Additional test import fixes needed for full compilation

fzowl committed:
Fix test compilation errors - imports and constructor signatures

- Update all VoyageAI test imports to use embeddings/text package
- Fix VoyageAIMultimodalEmbeddingsTaskSettings constructor calls (add truncation param)
- Fix VoyageAIMultimodalEmbeddingsServiceSettings constructor calls (add embeddingType param)
- Update test files: VoyageAIServiceTests, VoyageAIActionCreatorTests, VoyageAIEmbeddingsActionTests
- Update test files: VoyageAIEmbeddingsRequestTests, VoyageAIRequestTests
- Update test files: VoyageAIMultimodalEmbeddingsTaskSettingsTests, VoyageAIMultimodalEmbeddingsModelTests
- Update test files: VoyageAIMultimodalEmbeddingsRequestTests

Remaining: 3 compilation errors to fix

fzowl committed:
Fix remaining test compilation errors

- Fix VoyageAIMultimodalEmbeddingsModel constructor call in response tests
- Fix VoyageAIServiceTests multimodal model creation (add taskSettings param)
- Fix VoyageAIContextualEmbeddingsTaskSettings ambiguous constructor
- Cast SecretSettings to DefaultSecretSettings where required

fzowl committed:
Remove old external/voyageai files - moved to services/voyageai
